src/tls.c

```diff
@@ -10709,6 +11427,12 @@ static int TLSX_Write(TLSX* list, byte* output, byte* semaphore,
                 break;
 
 #endif /* WOLFSSL_DTLS_CID */
+#if defined(HAVE_ECH)
+            case TLSX_ECH:
+                offset += ECH_WRITE((WOLFSSL_ECH*)extension->data,
```

Comment:
With injected malloc failures, this fails due to ECH_WRITE returning a negative MEMORY_E (-125) value when the wc_InitRng fails.